src/commands/link.js

```diff
@@ -1,16 +1,54 @@
 // @flow
+import Project from '../Project';
+import * as yarn from '../utils/yarn';
+import * as logger from '../utils/logger';
 import * as options from '../utils/options';
 import { BoltError } from '../utils/errors';
+import * as messages from '../utils/messages';
 
-export type LinkOptions = {};
+function getWorkspaceMap(workspaces) {
+  let workspaceMap = new Map();
+
+  for (let workspace of workspaces) {
+    workspaceMap.set(workspace.pkg.config.getName(), workspace);
+  }
+
+  return workspaceMap;
+}
+
+export type LinkOptions = {|
+  cwd?: string,
+  packagesToLink: ?Array<string>
+|};
 
 export function toLinkOptions(
   args: options.Args,
   flags: options.Flags
 ): LinkOptions {
-  return {};
+  return {
+    cwd: options.string(flags.cwd, 'cwd'),
+    packagesToLink: args
+  };
 }
 
 export async function link(opts: LinkOptions) {
-  throw new BoltError('Unimplemented command "link"');
+  let cwd = opts.cwd || process.cwd();
+  let packagesToLink = opts.packagesToLink;
+  let project = await Project.init(cwd);
+  let workspaces = await project.getWorkspaces();
+  let workspaceMap = getWorkspaceMap(workspaces);
+
+  if (packagesToLink && packagesToLink.length) {
+    packagesToLink.forEach(async packageToLink => {
+      if (workspaceMap.has(packageToLink)) {
+        logger.warn(messages.linkInternalPackage(packageToLink));
+      } else {
+        await yarn.link(cwd, packageToLink);
+      }
+    });
```

Comment:
Can you write it like this?

```js
await Promise.all(packagesToLink.map(async packageToLink => {
  // ...
}));
```

Right now `link()` will resolve before all the packages are linked because nothing is `await`-ing the promises being created inside the `forEach`